lib/puppet/configurer.rb

```diff
@@ -44,6 +44,12 @@ def init_storage
     end
   end
 
+  def init_http_pool_monitoring(report, http_pool)
+    if http_pool.is_a? Puppet::Network::HTTP::Pool
+      http_pool.report = report
```

Comment:
I haven't done it because I thought it's not as lightweight as just assigning a attribute of the `Puppet::Network::HTTP::Pool` object. 
For the obsolete reference, we could just reassign `@report` to `nil` when needed